pkg/sbom/cyclonedx/marshal.go

```diff
@@ -345,6 +350,17 @@ func (*Marshaler) advisories(refs []string) *[]cdx.Advisory {
 	return &advs
 }
 
+// trimNonUrlInfo returns first valid URL.
+func trimNonUrlInfo(ref string) string {
+	ss := strings.Split(ref, " ")
```

Comment:
We just want to take the first block. `strings.Cut` seems enough.

```suggestion
	s, _, _ := strings.Cut(ref, " ")
```

Reply:
What about case when `ref` starts with extra characters?
I haven't seen this case, but I think it's possible.
(e.g. `(2.35-TRIAL) https://github.com/andk/cpanpm/commit/9c98370287f4e709924aee7c58ef21c85289a7f0`)

Reply:
Yes, I thought about it and didn't think we could handle all edge cases. But it's still better to support more edge cases. Let's keep it as is.